File: Entities/QuestionTypes/TextQuestion.cs
﻿using System.ComponentModel.DataAnnotations;
using Core.Entities;

namespace Entities.QuestionTypes
{
    public class TextQuestion : IEntity
    {
        [Key]
        public int QuestionId { get; set; }
        public bool IsRequired { get; set; }
        public bool IsMaxCharacters { get; set; }
        public int MaxCharacters { get; set; }
    }
}

File: WebUI/UI/src/app/datas/models/question-types/text-question.ts
export class TextQuestion {
  constructor(questionId: number, isRequired: boolean, isMaxCharacters: boolean
    ) {
    this.questionId = questionId;
    this.isRequired = isRequired;
    this.isMaxCharacters = isMaxCharacters;
  }

  questionId: number;
  isRequired: boolean;
  isMaxCharacters: boolean;
  maxCharacters: number;
}
